### DISCUSSION TITLE
OS X Sonoma

### DISCUSSION THREAD

**Discussion by DamianoMeacci** [lathoub/Arduino-AppleMIDI-Library#167] [Q&A]
Hi, I am experiencing issues with a project of mine based on the Teensy 4.1 and the Apple operating systems Sonoma. I found that the same issue occurs with the example Teensy41_NoteOnOffEverySec: the connection between the Teensy and the Mac seems to work correctly, but the computer does not receive any data. However, I do not encounter any problems with Windows computers, and the program correctly receives the notes. 
Has anyone encountered the same problem? Do you have any suggestions? 
Thanks.

**Comment by lathoub** [comment]
Interesting, i will check later today. Thx for signaling.

Have you tried using WireShark to see if the negotiation is happening correctly?

**Reply by DamianoMeacci** [reply]
No, I didn't try, I will check with WireShark later today.
To give you all the information in the Teensy41_NoteOnOffEverySec I change only one code line:

Ethernet.begin(mac);

I substitute with:

byte ip[] = {192, 168, 0, 70};
Ethernet.begin(mac, ip);

**Comment by DamianoMeacci** [comment]
Good morning, I apologize but I think I have understood that the problem is related to something in the operating system: if I reload the sketch with a different network configuration (especially with a different MAC address), the Apple computer refuses to recognize the device until I restart the computer. Therefore, I managed to get the Teensy41_NoteOnOffEverySec example to work by restarting the operating system with the sketch already loaded on the Teensy. 
In a more complex version of the sketch, I also implemented Bonjour. This version works very well on Windows. Trying to get this version to work, I realized that the Apple computer uses the peripheral only if, even though it sees the Teensy device listed in the directory, I manually add a peer with the same configuration as the one already present in the "directory" (IP address and port). 
If the Apple system goes to sleep or shuts down, it does not always reconnect directly to the Teensy device; often, I must remove it from the session and re-add it. This does not happen on Windows. On Windows, sometimes it takes a bit of time (10-15 seconds) to reconnect, but it does so on every occasion.
At this moment, I am in difficulty because it is not exactly clear to me what is causing these malfunctions and if there is a way to ensure that the system can be stable on Apple as it is on Windows. I will try to check if WireShark gives me any indications. If anyone has any ideas to suggest, they would be welcome. 
Thank you.
